seml/observers.py

```diff
@@ -75,3 +96,150 @@ def create_neptune_observer(project_name, api_token=None,
     neptune_obs = NeptuneObserver(api_token=api_token, project_name=project_name, source_extensions=source_extensions)
     return neptune_obs
+
+
+class MattermostObserver(RunObserver):
+    """
+    Based on Sacred's Slack observer: https://github.com/IDSIA/sacred/blob/master/sacred/observers/slack.py
+    Sends a message to Mattermost upon completion/failing of an experiment.
+    """
+
+    @classmethod
+    def from_config(cls, filename):
+        """
+        Create a MattermostObserver from a given configuration file.
+
+        The file can be in any format supported by Sacred
+        (.json, .pickle, [.yaml]).
+        It has to specify a ``webhook_url`` and can optionally set
+        ``bot_name``, ``icon``, ``completed_text``, ``interrupted_text``, and
+        ``failed_text``.
+        """
+        return cls(**load_config_file(filename))
+
+    def __init__(
+        self,
+        webhook_url,
+        channel=None,
+        bot_name="sacredbot",
+        icon=":angel:",
+        completed_text=None,
+        interrupted_text=None,
+        failed_text=None,
+    ):
+        """
+        Create a Sacred observer that will send notifications to Mattermost.
+        Parameters
+        ----------
+        webhook_url: str
+            The webhook for the bot account.
+        channel: str
+            The channel to which to send notifications. To send direct messages, set to @username.
+        bot_name: str
+            The name of the bot.
+        icon: str
+            The icon of the bot.
+        completed_text: str
+            Text to be sent upoon completion.
+        interrupted_text: str
+            Text to be sent upon interruption.
+        failed_text: str
+            Text to be sent upon failure.
+        """
+        self.webhook_url = webhook_url
+        self.bot_name = bot_name
+        self.icon = icon
+        self.completed_text = completed_text or (
+            ":white_check_mark: *{experiment[name]}* "
+            "completed after _{elapsed_time}_ with result=`{result}`"
+        )
+        self.interrupted_text = interrupted_text or (
+            ":warning: *{experiment[name]}* " "interrupted after _{elapsed_time}_"
+        )
+        self.failed_text = failed_text or (
+            ":x: *{experiment[name]}* failed after " "_{elapsed_time}_ with `{error}`"
+        )
+        self.run = None
+        self.channel = channel
+
+    def started_event(
+        self, ex_info, command, host_info, start_time, config, meta_info, _id
+    ):
+        self.run = {
+            "_id": _id,
+            "config": config,
+            "start_time": start_time,
+            "experiment": ex_info,
+            "command": command,
+            "host_info": host_info,
+        }
+
+    def get_completed_text(self):
+        return self.completed_text.format(**self.run)
+
+    def get_interrupted_text(self):
```

Comment:
Doesn't only the user manually interrupt experiments? Do we want to notify for this?

Reply:
df7918c adds options to specify whether to notify upon certain events. 

In general, there are these types of events:

* `queued_event`: fired when queuing the experiment via `Sacred`. Since we do the queuing ourselves, I don't think this applies to us.
* `started_event`: fired when the experiment starts. Could be useful in case the cluster is jammed so that people know when their jobs are started?
* `heartbeat_event`: experiments send a heartbeat every 10s. We could use this to notify a user when their experiment is probably dead, e.g. after not receiving a heartbeat for 5 minutes or so?
* `completed_event`: fired upon successful completion.
* `interrupted_event`: fired upon interrupt
* `failed_event`: fired upon failure
* `resource_event`: fired when a resource is added to the run. I think this is irrelevant for us.
* `artifact_event`: fired when an artifact is added to the run. Also irrelevant for us imo.

So, potentially `started_event` and `hearbeat_event` could be relevant for us. What do you think?

Reply:
I would add `started_event`, and disable both `started_event` and `interrupted_event` by default.

How would we do the trick with heartbeat? We can't check if the program is dead within the program itself.

Reply:
Yes, you're right. I was thinking about having checking this in our Observer, but as you pointed out, if the program is dead, our observer is also dead :)

Reply:
> I would add `started_event`, and disable both `started_event` and `interrupted_event` by default.

done in e688fd6